browser/test/Mocks/oni.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * Mocks/oni.ts
+ *
+ * Implementations of test mocks and doubles,
+ * for Oni's API interface.
+ */
+
+import { EventEmitter } from "events"
+import * as OniApi from "oni-api"
+
+import { Automation } from "./../../src/Services/Automation"
+import { Colors } from "./../../src/Services/Colors"
+import { CommandManager } from "./../../src/Services/CommandManager"
+import { Configuration } from "./../../src/Services/Configuration"
+import { DiagnosticsDataSource } from "./../../src/Services/Diagnostics"
+import { EditorManager } from "./../../src/Services/EditorManager"
+import { InputManager } from "./../../src/Services/InputManager"
+import { ThemeManager } from "./../../src/Services/Themes/ThemeManager"
+import { MenuManager } from "./../../src/Services/Menu"
+import { Process } from "./../../src/Plugins/Api/Process"
+import { StatusBar } from "./../../src/Services/StatusBar"
+import { Workspace } from "./../../src/Services/Workspace"
+import { WindowManager } from "./../../src/Services/WindowManager"
+
+import { MockPerformance } from "./Performance"
+
+export class OniMock extends EventEmitter implements OniApi.Plugin.Api {
```

Comment:
It's really cool you're thinking about this! Per the comments - I wonder if we will actually need to mock this completely, though?

In the case of a CiTest, we have a 'real' `Oni` api object available - for validating the end-to-end.

For a unit test, I'm thinking it might be preferable to use the specific mocks that the class under test needs - like instead of passing the entire API object, we pass a mock editormanager/windowmanager/colors. This helps make the dependencies explicit and minimizes the surface area.